contribute/design/1.x/apis/cstorvolume_policy.md

```diff
@@ -0,0 +1,320 @@
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// CStorVolumeConfigList is a list of CStorVolumeConfig resources
+type CStorVolumeConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []CStorVolumeConfig `json:"items"`
+}
+```
+
+# Changes to existing types
+
+- No changes
+
+# Volume Policy Schema Proposal
+
+## `CStorVolumePolicy` APIs
+
+```go
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// CStorVolumePolicy describes a configuration required for cstor volume
+// resources
+type CStorVolumePolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines a configuration info of a cstor volume required
+	// to provisione cstor volume resources
+	Spec   CStorVolumePolicySpec   `json:"spec"`
+	Status CStorVolumePolicyStatus `json:"status"`
+}
+
+// CStorVolumePolicySpec ...
+type CStorVolumePolicySpec struct {
+  // Provision represents volume provisioning configuration like replicaAffinity
+  // etc
+	Provision Provision `json:"provision"`
+
+  // TargetSpec represents configuration related to cstor target and its resources
+	Target TargetSpec `json:"target,omitempty"`
+
+  // ReplicaSpec represents configuration related to replicas resources
+	Replica ReplicaSpec `json:"replica,omitempty"`
+
+
+	// ReplicaPoolInfo holds the pool information of volume replicas.
+	// Ex: If volume is provisioned on which CStor pool volume replicas exist
+	ReplicaPoolInfo []ReplicaPoolInfo `json:"replicaPoolInfo,omitempty"`
+}
+
+// TargetSpec represents configuration related to cstor target and its resources
+type TargetSpec struct {
+	// QueueDepth sets the queue size at iSCSI target which limits the
+	// ongoing IO count from client
+	QueueDepth string `json:"queueDepth,omitempty"`
+
+	// Luworkers sets the number of threads that are working on above queue
+	LuWorkers int64 `json:"luWorkers,omitempty"`
+
+	// Monitor enables or disables the target exporter sidecar
+	Monitor bool `json:"monitor,omitempty"`
+
+	// ReplicationFactor represents maximum number of replicas
+	// that are allowed to connect to the target
+	ReplicationFactor int64 `json:"replicationFactor,omitempty"`
+
+	// Resources are the compute resources required by the cstor-target
+	// container.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// AuxResources are the compute resources required by the cstor-target pod
+	// side car containers.
+	AuxResources *corev1.ResourceRequirements `json:"auxResources,omitempty"`
+
+	// Tolerations, if specified, are the target pod's tolerations
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Affinity if specified, are the target pod's affinities
+	Affinity *corev1.PodAffinity `json:"affinity,omitempty"`
```

Comment:
n_: can this be called out as PodAffinity?

Reply:
done